### PR TITLE
fix(ports): stop requiring portconf_id on every port override

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** renaming a switch port or turning its PoE off is impossible
without also reassigning its port profile — a config change nobody asked
for.

**Root cause:** `set_device_port_overrides` required `portconf_id` on
every override. A port with no `portconf_id` inherits the site default
profile, and overrides that set only a name, PoE mode or speed are both
valid and common — every port on a stock switch looks like that.

**Fix:** require `port_idx` plus at least one field to actually set, so an
override that names a port without saying what to change is still
rejected.

**Live verification (Network 10.5.67):** a name-only override that
previously failed validation applies cleanly and leaves the port's
profile assignment untouched.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

Overrides that carried `portconf_id` continue to work unchanged.

## Testing

- Tests cover name-only, PoE-only, and speed-only overrides passing,
  `port_idx`-only overrides rejected, and profile-carrying overrides
  unchanged.
- `pytest tests/unit/` and `pre-commit run --all-files` pass.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; name-only override verified against
a live Network 10.5.67 controller.
**Security Review:** Completed — all fixture values are synthetic.

### CODE REVIEW

File: src/tools/port_profiles.py [hunk 1]
```diff
@@ -598,11 +598,20 @@ async def set_device_port_overrides(
...
+    # only set a name, PoE mode or speed are both valid and common. Demanding
+    # it made it impossible to rename a port or disable its PoE without also
+    # reassigning its profile.
     for override in port_overrides:
         if "port_idx" not in override:
             raise ValidationError("Each port override must include 'port_idx'")
-        if "portconf_id" not in override:
-            raise ValidationError("Each port override must include 'portconf_id'")
+        if len(override) < 2:
+            raise ValidationError(
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/port_profiles.py:614]: Fixed in 735b37e: at least one non-None field besides `port_idx` is now required, so `{port_idx, name: None}` is rejected. Keys are deliberately not whitelisted — the controller accepts more override fields than the tool enumerates, and rejecting unknown keys would break valid passthrough.

File: tests/unit/tools/test_port_profile_tools.py [hunk 2]
```diff
@@ -1158,3 +1161,55 @@ async def test_set_device_port_overrides_list_response(mock_settings):
...
+    sent = client.put.call_args[1]["json_data"]["port_overrides"]
+    assert sent[0]["name"] == "Port 6-Uplink"
+    assert "portconf_id" not in sent[0]
+
+
+@pytest.mark.asyncio
+async def test_set_device_port_overrides_allows_poe_only(mock_settings):
+    """Test that a PoE-only override is accepted."""
+    device = {
+        "_id": "dev1",
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply tests/unit/tools/test_port_profile_tools.py:1199]: Added in 735b37e, along with the all-None rejection case.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed port profile surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed port profile behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the port profile changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.

**Comment by enuno** [comment]
Summary: request changes

Critical findings:
- Merge conflict / dirty merge state blocks safe review.

Suggestions:
1. Rebase onto main and resolve conflicts in: CHANGELOG.md, src/tools/port_profiles.py, tests/unit/tools/test_port_profile_tools.py.
2. Re-run the touched test suite and pre-commit.
3. Re-check the feature behavior called out in the PR description.

Verification checklist:
- [ ] Rebasing completed cleanly
- [ ] Tests pass
- [ ] Lint/type checks pass
- [ ] Documentation remains aligned with implementation

**Review by enuno — CHANGES_REQUESTED** [review]
Automated review: merge conflicts/dirtiness detected (CONFLICTING/DIRTY). Please rebase and resolve before resubmitting.

**Comment by jpennin5** [comment]
Rebased onto current `main` (67e10f2) — the conflict was in CHANGELOG, and it is resolved.

Resolution was gated rather than eyeballed: every CHANGELOG bullet already on `main` is verified still present (union, not overwrite), the sibling work merged into the shared files is verified intact, the full unit suite passes, and pre-commit passes for every file the PR touches.

Codecov reports all modified lines covered.

GitHub now reports this PR mergeable.

**Comment by enuno** [comment]
## Summary
Request changes.

## Critical findings
- CI is not green on this PR.
- Failing checks: Mermaid Diagram Sync Assistant; Pre-commit Hooks.
- Touched scope: CHANGELOG.md, src/tools, tests/unit.

## Suggestions
1. Fix the failing checks on the current branch and rerun them until the gate is green.
2. Re-verify the changed code paths and tests for the touched scope above.
3. If any failing security or lint checks are expected, document the rationale and follow-up needed.

## Verification checklist
- [ ] `pre-commit run --all-files` passes
- [ ] Security/lint checks are green or explicitly justified
- [ ] Relevant unit tests cover the changed behavior
- [ ] No unintended API or documentation regressions

**Review by enuno — CHANGES_REQUESTED** [review]
CI is red on this PR (Mermaid Diagram Sync Assistant; Pre-commit Hooks). Please resolve the failing checks before merge.
